kpops/components/base_components/pipeline_component.py

```diff
@@ -75,6 +75,20 @@ def __init__(self, **kwargs) -> None:
         self.set_input_topics()
         self.set_output_topics()
 
+    def get_input_topics(self) -> list[str] | None:
+        """
+        Get all the input topics
+        """
+
+    def get_extra_output_topics(self) -> dict[str, str] | None:
+        ...
+
+    def get_extra_input_topics(self) -> dict[str, list[str]] | None:
```

Comment:
what is the meaning of the dict keys and values?

Reply:
it is the same as the set extra topics. You are getting the roles and the topics with that role.

Reply:
could you add a docstring to these new methods, please?

Reply:
@sujuka99 is this the correct docstring format?

Reply:
For a single-line docstr we use the following format:
```python
"""Get extra output topics list from config."""
```

If more than 1 line is needed:
```python
"""Summary goes here

This is the description. Leave 1 empty line between title and description.
The description can be as long as your heart desires
"""
 ```

Reply:
I think now should be fixed :)